### DISCUSSION TITLE
WPF - CommandManager

### DISCUSSION THREAD

**Discussion by CROprogrammer** [MarimerLLC/csla#3320] [Questions] [answered]
I'm upgrading WPF project that uses CSLA 3.6.0 to 6.2.2.
Is there any way to use someting like this in CSLA 6.2.2:

`<csla:CslaDataProvider x:Key="RoleList"
                           ObjectType="{x:Type proj:Roles}"
                           FactoryMethod="Get"
                           IsAsynchronous="False"
                           DataChanged="DataChanged"
                           ManageObjectLifetime="True" />`

`<Button 
              Command="ApplicationCommands.Undo" 
              CommandTarget="{Binding Source={StaticResource RoleList}, Path=CommandManager, BindsDirectlyToSource=True}"
              HorizontalAlignment="Left" IsCancel="True">Cancel</Button>`

I'm trying to get the Undo command, but when I use CslaDataProvider I have to provide factory method which is not use in 6.2.2. How do I solve this?

**Comment by rockfordlhotka** [comment, marked as answer]
@CROprogrammer that's a good question.

It is unlikely that we can resolve this in XAML, because there's no longer a fixed static method name like in the past. It is probably something that'll need to be done via code.

I don't have time to dig into it just this second, but will look when I get time. There might already be a way to invoke a method via code (like in the form load). If not, we'll need to add something like what is in the Blazor viewmodel class for invoking code to create/fetch the domain object via the data portal.

**Reply by CROprogrammer** [reply]
Thank you. It is possible to write code that does that, but I just wanted to make sure that this way is no longer possible.
